### PR TITLE
PT-157931784 Sophia TTL type -- relative oracle extension TTL

### PR DESCRIPTION
**Pull request by lucafavatella**
TODO:
* [x] Fix tests in aecontract_SUITE using domain-specific language for defining test cases and state machine for executing them. @UlfNorell Could you please help adapting those?

### CODE REVIEW

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 1]
```diff
@@ -758,7 +758,7 @@ sophia_oracles(_Cfg) ->
     none              = ?call(call_contract, Acc, Ct, getAnswer, {option, word}, {CtId, QId}),
     {}                = ?call(call_contract, Acc, Ct, respond, {tuple, []}, {CtId, QId, 0, 4001}),
     {some, 4001}      = ?call(call_contract, Acc, Ct, getAnswer, {option, word}, {CtId, QId}),
-    {}                = ?call(call_contract, Acc, Ct, extendOracle, {tuple, []}, {Ct, 0, FixedTTL(TTL + 10)}),
+    {}                = ?call(call_contract, Acc, Ct, extendOracle, {tuple, []}, {Ct, 0, RelativeTTL(TTL + 10)}),
```
Comment (UlfNorell) [inline apps/aecontract/test/aecontract_SUITE.erl]: Should be `RelativeTTL(10)`

**Comment by hanssv** [comment]
Though dialyzer isn't happy...

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 2]
```diff
@@ -851,8 +852,8 @@ ttl_scenario_create_and_extend(Start0, Extend0, Stop0) ->
       || Start <- List(Start0), Extend <- List(Extend0), Stop <- List(Stop0),
          TTLo <- [ T || Extend == false, T <- ttls(Start, [Stop]) ] ++
                  [ T || Extend /= false, T <- ttls(Start, [Extend + 5]) ],
-         TTLe <- [ 0 || Extend == false ] ++
-                 [ T || Extend /= false, T <- ttls(ttl_height(Start, TTLo), [Stop]) ] ].
+         TTLe = {delta, _} <- [ 0 || Extend == false ] ++
```
Comment (lucafavatella) [inline apps/aecontract/test/aecontract_SUITE.erl]: I see this filters only the ones matching `{delta, _}` hence leaving out the ones matching `{block, _}`. Ok.

Comment (UlfNorell) [reply apps/aecontract/test/aecontract_SUITE.erl]: Oh, but the `[ 0 || Extend == false ]` will never match, so we won't get any test cases that doesn't extend. This should be `[ {delta, 0} || Extend == false ]` instead.

**Comment by lucafavatella** [comment]
This shall be ready now. CI running.

**Pull request merged**